File: samples/Cpp/XAMLBehaviorsSampleCpp/ContentDialogPopUp.cpp
#pragma once

#include "pch.h"
#include "ContentDialogPopUp.h"

using namespace Platform;
using namespace XAMLBehaviorsSampleCpp;

Object^ ContentDialogPopUp::Execute(Platform::Object^ sender, Platform::Object^ parameter)
{
	samplecd = ref new ContentDialogSample();
	samplecd->ShowAsync();
	return nullptr;
}

File: samples/CS/XAMLBehaviorsSample/ContentDialogPopUp.cs
using Microsoft.Xaml.Interactivity;
using System;
using Windows.UI.Xaml;

namespace XAMLBehaviorsSample;

class ContentDialogPopUp : DependencyObject, IAction
{
    ContentDialogSample samplecd;

    public object Execute(object sender, object parameter)
    {
        samplecd = new ContentDialogSample();
        ShowCD();
        return null;
    }

    public async void ShowCD()
    {
        await samplecd.ShowAsync();
    }
}
